src/views/Home/Charts/TimelineChart/index.tsx

```diff
@@ -0,0 +1,273 @@
+import {
+    useRef,
+    useCallback,
+    useMemo,
+    useState,
+    Fragment,
+} from 'react';
+import { listToMap, _cs, bound } from '@togglecorp/fujs';
+import useSizeTracking from '#hooks/useSizeTracking';
+import {
+    getBounds,
+    getPathData,
+    getScaleFunction,
+} from '#utils/chart';
+
+import ChartPoint from './ChartPoint';
+import styles from './styles.module.css';
+
+const X_AXIS_HEIGHT = 20;
+const Y_AXIS_WIDTH = 10;
+const CHART_OFFSET = 10;
+
+const chartMargin = {
+    left: 2 * Y_AXIS_WIDTH + CHART_OFFSET,
+    top: CHART_OFFSET,
+    right: CHART_OFFSET * 2,
+    bottom: 2 * X_AXIS_HEIGHT + CHART_OFFSET,
+};
+
+const getFormattedKey = (dateFromProps: string | Date) => {
+    const date = new Date(dateFromProps);
+    return `${date.getFullYear()}-${date.getMonth()}`;
+};
+
+interface Props<K> {
+    className?: string;
+    dataKeys: K[];
+    timePoints: Date[];
+    valueSelector: (dataKey: K, date: Date) => number;
+    classNameSelector: (dataKey: K) => string;
+    activePointKey: string;
+    onTimePointClick: (dateKey: string) => void;
+    xAxisFormatter: (date: Date) => string;
+}
+
+function TimelineChart<const K extends string>(props: Props<K>) {
+    const {
+        className,
+        dataKeys,
+        timePoints,
+        valueSelector,
+        classNameSelector,
+        activePointKey,
+        onTimePointClick,
+        xAxisFormatter,
+    } = props;
+
+    const [hoveredPointKey, setHoveredPointKey] = useState<string | undefined>();
+
+    const svgRef = useRef<SVGSVGElement>(null);
+    const bounds = useSizeTracking(svgRef);
+
+    const [
+        chartPoints,
+        yAxisPoints,
+    ] = useMemo(
+        () => {
+            if (!bounds) {
+                return [
+                    [],
+                    [],
+                ];
+            }
+
+            const timePointsWithData = timePoints.map((timePoint) => {
+                const dataValues = dataKeys.map((dataKey) => ({
+                    key: dataKey,
+                    value: valueSelector(dataKey, timePoint),
+                }));
+
+                const valueList = dataValues.map((value) => value.value);
+                const maxValue = valueList.length === 0 ? 0 : Math.max(...valueList);
```

Comment:
We might get "-Infinity" as value, do we handle that?

Reply:
lol